### ISSUE TITLE
How to adopt code for a new LLM? 

### ISSUE THREAD

**Issue by fabriceyhc** [yepengliu/Adaptive-Watermark#2]
Thanks for publishing your code for this watermarker!

Unfortunately OPT-6.7B is pretty weak and I want to try watermarking with Llama-3-70B. However, I understand that I will need to train a new "Semantic Mapping (SM)" model (which your code seems to call a TransformModel). Can you publish the training script you used (e.g. custom loss function, dataset, etc) to arrive at the `transform_mapping_list` and `transform_model`? 

More generally, how can your code be adopted to use with a new LLM?

**Comment by yepengliu** [comment]
Thanks for your interest in our work. I have updated the code to make it automatically accommodate any LLM and generate `transform_mapping_list` without the need to train a new SMM, just need to specify the size of tokenizer. BTW, the training script of SMM is also uploaded. Feel free to let me know if you have any further problem.
